### PR TITLE
Countdowns

### PR DESCRIPTION
**Pull request by The1nstructor**
# Introducing Countdowns

**This branch works on the state after master-merging #25 the new nipple actions.**

TODO:
- [x] advanced Edging
  _Implementing advanced edging functionality that spices up middle-late gameplay._

- [x] advanced Orgasm
  _Implementing advanced orgasm functionality that spices up late gameplay due to multiple endings._

*EDIT* *will do advanced Ruin in future Branch*
>advanced Ruin
  _Implementing advanced ruin functionality that spices up late gameplay due to multiple endings._

I always tried to create inline documentation for what I code and what I read and reuse.

### CODE REVIEW

**Comment by The1nstructor** [comment]
I will @Mention you when I'm done here 😉I just created the PR so you all can overwatch my progress / provide feedback and to make you notice that I am already working on this.

File: src/game/actions/orgasm/orgasmInTime.js [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+ * maybe transparency 50% and fading out after 1-2 seconds of display
+ *
+ * When reaching Zero the user gets a very short amount of time to orgasm.
+ *
+ * @returns {Promise<void>}
+ */
+export const doOrgasmCountdown = async () => {
+  //TODO: Implement // I Need help @thefapinstructor I intend a implementation like:
+  // for (let i = 10; i >= 0; i--) {
+  //   displayNumber(i);
```
Comment (The1nstructor) [inline src/game/actions/orgasm/orgasmInTime.js]: @thefapinstructor Could you please help me out with implementing that displayNumber(number) function? 

I don't completely understand the game page and don't want to mess up stuff.

> I am thinking of big white letters in the middle of the screen over the videos/gifs/pictures.
maybe transparency 50% and fading out after 1-2 seconds of being displayed.

```js
for (let i = 10; i >= 0; i--) {
  displayNumber(i);
  const time = getRandomInclusiveInteger(1, 5);
  await delay(time * 1000);
}
```

**Comment by The1nstructor** [comment]
btw. travis currently fails because of unused vars.

**Comment by The1nstructor** [comment]
Hey @thefapinstructor,

# I am **done** here. 

## This first *Major Update* on *Countdowns* is ready for **Master Merge**

However, I will further work on the advanced topics in Countdown (#27) topic in a new branch I will create soon. But for now, I may also already share my work with ye all.

@thefapinstructor  I still would appreciate if you could assist me with that [Problem](https://github.com/thefapinstructor/fapinstructor/pull/42#discussion_r208988325) from above. But this does not affect the current game. 

so, nothing to do but to 🚢 ship this 😉 

Greetings
the1nstructor

**Comment by The1nstructor** [comment]
Hey @thefapinstructor,

# I am **done** here. 

## This first *Major Update* on *Countdowns* is ready for **Master Merge**

However, I will further work on the advanced topics in Countdown (#27) topic in a new branch I will create soon. But for now, I may also already share my work with ye all.

@thefapinstructor  I still would appreciate if you could assist me with that [Problem](https://github.com/thefapinstructor/fapinstructor/pull/42#discussion_r208988325) from above. But this does not affect the current game. 

so, nothing to do but to 🚢 ship this 😉 

Greetings
the1nstructor

**Comment by The1nstructor** [comment]
Hey @thefapinstructor,

# I am **done** here. 

## This first *Major Update* on *Countdowns* is ready for **Master Merge**

However, I will further work on the advanced topics from the Countdown (#27) topic in a new branch I will create soon. But for now, I may also already share my work with ye all.

@thefapinstructor  I still would appreciate if you could assist me with that [Problem](https://github.com/thefapinstructor/fapinstructor/pull/42#discussion_r208988325) from above. But this does not affect the current game. 

so, nothing to do but to 🚢 ship this 😉 

Greetings
the1nstructor

**Comment by The1nstructor** [comment]
Closes #6

**Comment by thefapinstructor** [comment]
@The1nstructor awesome work.  Things have been busy but I'll try to have a look at the problem this Saturday.  Thanks for your patience!

**Comment by The1nstructor** [comment]
Great! 

You might actually already merge this PR, then I could introduce the more advanced stuff in a new PR (only the advanced stuff needs the things mentioned in the problem 😉) so this patch is already working as standalone 😜 

You are welcome @thefapinstructor

**Comment by thefapinstructor** [comment]
@The1nstructor merging into master.  I noticed a couple of console warnings, and I think the orgasm section could be organized a bit cleaner but overall great work! 
I'll give that countdown timer a shot with my next merge this weekend.

**Comment by The1nstructor** [comment]
Thank you very much @thefapinstructor 

Yeah the orgasm part was and is quite difficult to manage. But I did not want to replace the old orgasm style so users who like the old style more can still use it. 

Also for the edging ladder I bet that there is a _nicer_ way to do it but the whole action iteration loop stuff was quite confusing 😅

But either way - do you mean that with cleaner or something else? I will try to improve that as soon as I find the time to work on Countdowns 2.0 😉 

I’ll also take a look at these warnings. Maybe you could paste the warnings in a new issue? 

So long and thanks for the help with the countdown timer with the fancy display-Style of the big numbers 😜

**Comment by The1nstructor** [comment]
Oh i think i get what you mean by cleaner. 

The website style is not looking as expected. Not like shown by my dev-server ... I will fix that ASAP.

**Pull request merged**